contracts/os/andromeda-kernel/src/execute.rs

```diff
@@ -713,6 +728,43 @@ pub fn unset_env(execute_ctx: ExecuteContext, variable: String) -> Result<Respon
         .add_attribute("variable", variable))
 }
 
+/// Validates the existing ID of a packet
+fn validate_id(
+    id: &str,
+    current_chain_id: &str,
+    current_index: u32,
+) -> Result<String, ContractError> {
+    // Split the ID into chain_id and index parts at first occurrence of '.'
+    let (chain_id, index_str) = id
```

Comment:
This should probably have 3 elements, `chain.block.index` as I think the transaction index is per block?